tests/e2e/nvidia-container-cli_test.go

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	libnvidiaContainerCliTestTemplate = `#!/bin/bash
+set -euo pipefail
+export DEBIAN_FRONTEND=noninteractive
+
+# Add Docker's official GPG key:
+apt-get update
+apt-get install -y ca-certificates curl apt-utils
+install -m 0755 -d /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+chmod a+r /etc/apt/keyrings/docker.asc
+
+# Add the repository to Apt sources:
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+  $(. /etc/os-release && echo "${UBUNTU_CODENAME:-$VERSION_CODENAME}") stable" | \
+  tee /etc/apt/sources.list.d/docker.list > /dev/null
+apt-get update
+
+apt-get install -y docker-ce docker-ce-cli containerd.io
+
+# start dockerd in the background
+dockerd &
+
+# wait for dockerd to be ready
+while ! docker info > /dev/null 2>&1; do
+    echo "Waiting for dockerd to be ready..."
+    sleep 1
+done
+
+# install nvidia-container-toolkit from NVIDIA_TOOLKIT_IMAGE using docker
+TEMP_DIR="/tmp/nvidia-container-toolkit"
+mkdir -p "$TEMP_DIR"
+rm -f /usr/bin/nvidia-container-runtime-hook
+ln -s "$TEMP_DIR/toolkit/nvidia-container-runtime-hook" /usr/bin/nvidia-container-runtime-hook
+
+docker run --rm -d --name nvidia-container-toolkit-packaging --entrypoint "/usr/bin/sleep" ${NVIDIA_TOOLKIT_IMAGE}-packaging 120
+docker cp nvidia-container-toolkit-packaging:/artifacts/packages/ubuntu18.04/amd64 ${TEMP_DIR}
+docker rm -f nvidia-container-toolkit-packaging
```

Comment:
What about something like: https://github.com/NVIDIA/nvidia-container-toolkit/blob/08b3a388e7b1d447e10d4c4d4a71dca29a98a964/hack/pull-packages.sh#L50-L58 instead? Then we don't have to worry about stopping the container again.

Reply:
done